### PR TITLE
Airflow Plugin

### PR DESCRIPTION
**Pull request by samhita-alla**
Signed-off-by: Samhita Alla <aallasamhita@gmail.com>

### CODE REVIEW

**Review by kumare3 — COMMENTED** [review]
Looks good to me. I did not review it closely. Will do in sometime

File: README.md [hunk 1]
```diff
@@ -1 +1,71 @@
...
+- Host(optional): The FlyteAdmin host. Defaults to localhost.
+- Port (optional): The FlyteAdmin port. Defaults to 30081.
+- Login (optional): `client_id`
+- Password (optional): `client_credentials_secret`
+- Extra (optional): Specify the `extra` parameter as JSON dictionary to provide additional parameters.
+  - `project`: The default project to connect to.
+  - `domain`: The default domain to connect to.
+  - `insecure`: Whether to use SSL or not.
+  - `command`: The command to execute to return a token using an external process.
+  - `scopes`: List of scopes to request.
```
Comment (kumare3) [inline README.md:28]: ideally you do not need this. But if you need for certain cases, then you should support all of these - https://github.com/flyteorg/flytekit/blob/195d26ab0cc03260bb96b828219fcf7e6ea768a1/flytekit/configuration/internal.py#L61

File: README.md [hunk 2]
```diff
@@ -1 +1,71 @@
...
+## Modules
+
+### [Flyte Operator](https://github.com/flyteorg/airflow-provider-flyte/blob/main/flyte_provider/operators/flyte.py)
+
+The `FlyteOperator` requires a `flyte_conn_id` to fetch all the connection-related
+parameters that are useful to instantiate `FlyteRemote`. Also, you must give a
+`launchplan_name` to trigger a workflow, or `task_name` to trigger a task; you can give a
+handful of other values that are optional, such as `project`, `domain`, `max_parallelism`,
+`raw_data_prefix`, `assumable_iam_role`, `kubernetes_service_account`, `labels`, `annotations`,
+`secrets`, `notifications`, `disable_notifications`, `oauth2_client`, `version`, and `inputs`.
```
Comment (kumare3) [inline README.md:40]: also now supported `interruptible`

File: flyte_provider/example_dags/example_flyte.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    catchup=False,
+) as dag:
+    # do not wait for the execution to complete
+    flyte_execution = FlyteOperator(
+        task_id="flyte_task",
+        flyte_conn_id="flyte_conn_example",
+        project="flytesnacks",
+        domain="development",
+        launchplan_name="core.basic.lp.my_wf",
+        assumable_iam_role="default",
```
Comment (kumare3) [inline flyte_provider/example_dags/example_flyte.py:24]: lets drop assumable_iam_role? as we have mostly migrated to using serviceaccount only right?

**Review by kumare3 — APPROVED** [review]
This is awesome +1

**Comment by kumare3** [comment]
couple small comments

File: README.md (hunk 2 above)
Comment (samhita-alla) [reply README.md:40]: `interruptibe` needs to be given in the task metadata, right? I don't think we can send `interruptible` when executing them through FlyteRemote.

**Review by pingsutw — COMMENTED** [review]
Thanks, this is awesome.
It seems like we need to add a couple of Github Action workflows to run the tests.
I can help after merging this PR.

File: README.md (hunk 1 above)
Comment (samhita-alla) [reply README.md:28]: I've already added the `PlatformConfig`-related params. Let me also add the other config params: https://github.com/flyteorg/flytekit/blob/195d26ab0cc03260bb96b828219fcf7e6ea768a1/flytekit/configuration/__init__.py#L483.

**Review by pingsutw — APPROVED** [review]
LGTM, thanks

**Pull request merged**